### ISSUE TITLE
Epic: Refactor website header to provide accessible navigation

### ISSUE THREAD

**Issue by roslynwythe** [hackforla/website#4300]
### Dependencies
These issues have been created from the epic:
- [x] Issue 1: #4301 
  - Referring to the W3C sample code for navigation menu buttons (see Resources) update the HTML and JavaScript to do the following:
    - make explicit the role and name of the hamburger icon/button using WAI-ARIA attributes, so that a screen reader is informed about the purpose of the button. 
    - add JavaScript to dynamically update the WAI-ARIA label "aria-expanded" so that a screen reader can announce whether the menu is expanded.  Alternatively, create two separate `<button>` elements, one for displaying the menu and the other for collapsing the menu, and mark each with an appropriate WAI-ARIA label.  
    - assign the `taborder` attribute to menu items so they are usable using keyboard navigation.  
    - recommend if it is necessary to reformat the HTML in order to place the navigation button/menu within a `<hav>` element, or if `role="nav"` should be added to another element.  
- [ ] Issue 2: Enhance accessibility of the HfLA logo/Homepage link
  - Refactor HTML so that the HfLA logo (? and the Donate link) are contained within a `<nav>` element.  
  - Improve "screen reader only" text for the logo or alternatively provide an explicit "Home" link.  
- [ ] Issue 3: Update the wiki page https://github.com/hackforla/website/wiki/How-to-make-SVGs-and-other-images-WCAG-compliant with guidance specific to button icons. 

### Overview
This epic addresses several problems navigating the site using a screen reader. 

### Details for Issue 1
The following issues prevent screen reader users from being able to navigate the website for mobile screen sizes:
1. On page load there is no accessible name or role given to the "hamburger" icon/button to indicate that activating the button will display the primary navigation menu.   The screen reader announces the button as "button list item" and so many users will skip over the button and lack access to the navigation links.   
2. If the mobile navigation menu links are displayed (when the "hamburger" navigation menu button is activated), the menu item links are not keyboard accessible because they are not in the tab order.  This is a result of the dynamic placement of the menu items. 

### Details for Issue 2
1.  In both mobile and desktop views, the HfLA logo is not enclosed in the `<nav>` element and is not identified explicitly as representing the Home page.  Currently it is identified to the screen reader user (using `sr-only` text) as "Hack for LA internal link".  Options include changing the` sr-only` text to "Hack for LA Homepage",  adding aria-labels/roles to clarify, or adding an explicit Home link in addition to the clickable logo. 
2. In both mobile and desktop views, the Donate link is not enclosed in the `<nav>` element, 

### Resources
- [Header Include](https://github.com/hackforla/website/blob/gh-pages/_includes/header.html) 
- [main nav Include](https://github.com/hackforla/website/blob/gh-pages/_includes/main-nav.html)
- [hamburger nav Include](https://github.com/hackforla/website/blob/gh-pages/_includes/hamburger-nav.html)
- [W3C - Menu Button Example](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/)
- [ARIA - Navigation Role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/navigation_role)

### Recommended Action Items
- [ ] Make a new issue
- [ ] Discuss with team
- [ ] Let a Team Lead know

### Potential solutions [draft]
- Modify #4301 to apply aria labels as described in https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/
- Research: which links should be enclosed by the `<nav>` element.  
- Refactor code to create a `<nav>` element that includes both mobile and wider screens,  Requires updates to include files: header.html, main-nav.html and hamburger-nav.html and associated CSS OR Create a separate `<nav>` element for mobile view. 
- In either case,

**Comment by ExperimentsInHonesty** [comment]
@roslynwythe Friendly reminder to add a size label to this issue

**Comment by LRenDO** [comment]
Hi @roslynwythe! Checking with you to make sure this issue is in the correct column of the board since it never went through the backlog and hasn't had an update in a while. Thanks!

**Comment by HackforLABot** [comment]
@roslynwythe

Please add update using the below template (even if you have a pull request). Afterwards, remove the '2 weeks inactive' label and add the 'Status: Updated' label.
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures (optional): "Add any pictures of the visual changes made to the site so far."

If you need help, be sure to either: 1) place your issue in the `Questions/In Review` column of the Project Board and ask for help at your next meeting, 2) put a "Status: Help Wanted" label on your issue and pull request, or 3) put up a request for assistance on the #hfla-site channel. Please note that including your questions in the issue comments- along with screenshots, if applicable- will help us to help you. [Here](https://github.com/hackforla/website/issues/1619#issuecomment-897315561) and [here](https://github.com/hackforla/website/issues/1908#issuecomment-877908152) are examples of well-formed questions.

<sub>You are receiving this comment because your last comment was before Tuesday, August 13, 2024 at 12:14 PM PST.</sub>

**Comment by HackforLABot** [comment]
@roslynwythe

Please add update using the below template (even if you have a pull request). Afterwards, remove the '2 weeks inactive' label and add the 'Status: Updated' label.
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures (optional): "Add any pictures of the visual changes made to the site so far."

If you need help, be sure to either: 1) place your issue in the `Questions/In Review` column of the Project Board and ask for help at your next meeting, 2) put a "Status: Help Wanted" label on your issue and pull request, or 3) put up a request for assistance on the #hfla-site channel. Please note that including your questions in the issue comments- along with screenshots, if applicable- will help us to help you. [Here](https://github.com/hackforla/website/issues/1619#issuecomment-897315561) and [here](https://github.com/hackforla/website/issues/1908#issuecomment-877908152) are examples of well-formed questions.

<sub>You are receiving this comment because your last comment was before Tuesday, August 27, 2024 at 12:04 AM PST.</sub>

**Comment by HackforLABot** [comment]
@roslynwythe

Please add update using the below template (even if you have a pull request). Afterwards, remove the '2 weeks inactive' label and add the 'Status: Updated' label.
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures (optional): "Add any pictures of the visual changes made to the site so far."

If you need help, be sure to either: 1) place your issue in the `Questions/In Review` column of the Project Board and ask for help at your next meeting, 2) put a "Status: Help Wanted" label on your issue and pull request, or 3) put up a request for assistance on the #hfla-site channel. Please note that including your questions in the issue comments- along with screenshots, if applicable- will help us to help you. [Here](https://github.com/hackforla/website/issues/1619#issuecomment-897315561) and [here](https://github.com/hackforla/website/issues/1908#issuecomment-877908152) are examples of well-formed questions.

<sub>You are receiving this comment because your last comment was before Tuesday, September 3, 2024 at 12:04 AM PST.</sub>

**Comment by HackforLABot** [comment]
@roslynwythe

Please add update using the below template (even if you have a pull request). Afterwards, remove the '2 weeks inactive' label and add the 'Status: Updated' label.
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures (optional): "Add any pictures of the visual changes made to the site so far."

If you need help, be sure to either: 1) place your issue in the `Questions/In Review` column of the Project Board and ask for help at your next meeting, 2) put a "Status: Help Wanted" label on your issue and pull request, or 3) put up a request for assistance on the #hfla-site channel. Please note that including your questions in the issue comments- along with screenshots, if applicable- will help us to help you. [Here](https://github.com/hackforla/website/issues/1619#issuecomment-897315561) and [here](https://github.com/hackforla/website/issues/1908#issuecomment-877908152) are examples of well-formed questions.

<sub>You are receiving this comment because your last comment was before Tuesday, September 10, 2024 at 12:04 AM PST.</sub>

**Comment by HackforLABot** [comment]
@roslynwythe

Please add update using the below template (even if you have a pull request). Afterwards, remove the '2 weeks inactive' label and add the 'Status: Updated' label.
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures (optional): "Add any pictures of the visual changes made to the site so far."

If you need help, be sure to either: 1) place your issue in the `Questions/In Review` column of the Project Board and ask for help at your next meeting, 2) put a "Status: Help Wanted" label on your issue and pull request, or 3) put up a request for assistance on the #hfla-site channel. Please note that including your questions in the issue comments- along with screenshots, if applicable- will help us to help you. [Here](https://github.com/hackforla/website/issues/1619#issuecomment-897315561) and [here](https://github.com/hackforla/website/issues/1908#issuecomment-877908152) are examples of well-formed questions.

<sub>You are receiving this comment because your last comment was before Tuesday, September 17, 2024 at 12:04 AM PST.</sub>

**Comment by HackforLABot** [comment]
@roslynwythe

Please add update using the below template (even if you have a pull request). Afterwards, remove the '2 weeks inactive' label and add the 'Status: Updated' label.
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures (optional): "Add any pictures of the visual changes made to the site so far."

If you need help, be sure to either: 1) place your issue in the `Questions/In Review` column of the Project Board and ask for help at your next meeting, 2) put a "Status: Help Wanted" label on your issue and pull request, or 3) put up a request for assistance on the #hfla-site channel. Please note that including your questions in the issue comments- along with screenshots, if applicable- will help us to help you. [Here](https://github.com/hackforla/website/issues/1619#issuecomment-897315561) and [here](https://github.com/hackforla/website/issues/1908#issuecomment-877908152) are examples of well-formed questions.

<sub>You are receiving this comment because your last comment was before Tuesday, October 8, 2024 at 12:05 AM PST.</sub>

**Comment by ExperimentsInHonesty** [comment]
- When issue 2 gets made, add it as a dependency for the audit issue #1347
